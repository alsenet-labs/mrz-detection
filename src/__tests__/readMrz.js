'use strict';

const path = require('path');

const Image = require('image-js').Image;

const { getMrz, readMrz } = require('../..');

it(
  'test the extraction of MRZ characters on an identity card',
  async () => {
    const img = await Image.load(path.join(__dirname, 'fixtures/id1.png'));
    let mrzImage = getMrz(img);
    // mrzImage.save('./test.jpg');
    let mrz = await readMrz(mrzImage);
    expect(mrz).toMatchSnapshot();
  },
  10000
);
