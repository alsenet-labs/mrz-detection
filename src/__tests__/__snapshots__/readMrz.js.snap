// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the extraction of MRZ characters on an identity card 1`] = `
Array [
  "IDAUT000009993<<<<<<<<<<<<<<<<",
  "7109094F111231AUT<<<<<<<<<<<<0",
  "MUSTERFRAU<<ISOLDE<<<<<<<<<<<<",
]
`;
